cmd/root.go

```diff
@@ -0,0 +1,234 @@
+}
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "vault-init",
+	Short: "Intitialise a specified instance of Vault",
+	Long: `This command makes a request to an instance of Vault at the specified Vault address
+to initialise the instance. This command currently assumes that auto-unseal has
+been setup to occur during Vault initialisation.
+
+Once it has received the token in the response from Vault, it will encrypt and
+store this token on S3 from where it can be used for authentication by entities that
+need to read from or write to the Vault instance.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		// make vault init request and get root token
+		fmt.Println("Initialising Vault...")
+		rootToken := initVault()
+		fmt.Println("Initialisation complete")
+
+		// AWS setup
+		region := os.Getenv("AWS_REGION")
+		sess := session.Must(session.NewSession(&aws.Config{Region: aws.String(region)}))
+		var kmsClient = kms.New(sess, aws.NewConfig().WithRegion(region))
+		uploader := s3manager.NewUploader(sess)
+
+		// encrypt tokens with AWS KMS
+		fmt.Println("Encrypting root token...")
+		encryptedToken, errE := kmsClient.Encrypt(&kms.EncryptInput{
+			KeyId: aws.String(fullKeyID(os.Args[1], os.Args[2], region)),
+			Plaintext: []byte(rootToken),
+		})
+		checkError(errE)
+		fmt.Println("Encryption complete.")
+
+		hostname, errH := os.Hostname()
+		if errH != nil {
+			panic(errH)
+		}
+
+		tokenFileName := hostname+ "_token"
+
+		writeToFile(tokenFileName, encryptedToken.CiphertextBlob)
+
+		// upload keys to S3
+		fmt.Println("Uploading encrypted token to S3...")
+		f := openFile(tokenFileName)
+
+		s3Result, errS3 := uploader.Upload(&s3manager.UploadInput{
+			Bucket: aws.String("encrypted-tokens"),
+			Key:    aws.String(tokenFileName),
+			Body:   f,
+		})
+		checkError(errS3)
+		fmt.Println("Encrypted token successfully uploaded to S3 at", s3Result.Location)
+	},
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func init() {
+	cobra.OnInitialize(initConfig)
+}
+
+
+// initConfig reads in config file and ENV variables if set.
+func initConfig() {
+	if cfgFile != "" {
+		// Use config file from the flag.
+		viper.SetConfigFile(cfgFile)
+	} else {
+		// Find home directory.
+		home, err := homedir.Dir()
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+
+		// Search config in home directory with name ".vault-init" (without extension).
+		viper.AddConfigPath(home)
+		viper.SetConfigName(".vault-init")
+	}
+
+	viper.AutomaticEnv() // read in environment variables that match
+
+	// If a config file is found, read it in.
+	if err := viper.ReadInConfig(); err == nil {
+		fmt.Println("Using config file:", viper.ConfigFileUsed())
+	}
```

Comment:
I don't think we need anything here since we're not reading configuration from anywhere. Should have been removed as part of addressing https://github.com/Intellection/vault-init/pull/1#discussion_r332846779.